File: test/scripts/28-cache-friendly-layouts/correct.js
const N = 500_000;
const DURATION_SEC = parseInt(process.env.DURATION_SEC || '60', 10);
const ids = Array.from({ length: N }, (_, i) => i);
let laps = 0;
const start = Date.now();
console.log('CORRECT: SoA');
function run() {
  if (Date.now() - start >= DURATION_SEC * 1000) {
    console.log('Stopped:', laps, 'laps');
    return;
  }
  let total = 0;
  for (let i = 0; i < ids.length; i++) total += ids[i];
  laps++;
  setImmediate(run);
}
run();
process.on('SIGINT', () => process.exit(0));

File: test/scripts/28-cache-friendly-layouts/correct.py
#!/usr/bin/env python3
"""
CORRECT: Struct of Arrays - iterating ids loads only ids.
Resume: "Use Cache-Friendly Memory Layouts"
"""
import os
import signal
import time
N = 500_000
DURATION_SEC = int(os.environ.get("DURATION_SEC", "60"))
stop = False

def main():
    global stop
    ids = list(range(N))
    print("CORRECT: SoA - only load ids")
    signal.signal(signal.SIGINT, lambda s, f: globals().__setitem__('stop', True))
    start = time.perf_counter()
    laps = 0
    while not stop and (time.perf_counter() - start) < DURATION_SEC:
        total = sum(ids)
        laps += 1
    print(f"Stopped: {laps} laps")

if __name__ == "__main__":
    main()
